source/http/httpsource.go

```diff
@@ -283,6 +283,10 @@ func (hs *httpSourceHandler) save(ctx context.Context, resp *http.Response, s se
 	}
 	fp := filepath.Join(dir, getFileName(hs.src.URL, hs.src.Filename, resp))
 
+	if err := os.MkdirAll(filepath.Dir(fp), 0600); err != nil {
```

Comment:
We should check if `filepath.Dir(fp) == ""` first, to avoid calling `os.MkdirAll("")` if there is no parent directory.